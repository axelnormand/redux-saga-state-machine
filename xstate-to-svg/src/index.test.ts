import { xstateToSvg } from './index';

describe('xstateToSvg', () => {
  it('...', () => {
    const description = {
      id: 'guards',
      initial: 'closed',
      states: {
        closed: {
          initial: 'idle',
          states: {
            idle: {},
            error: {},
          },
          on: {
            OPEN: [
              { target: 'opened', cond: (extState) => extState.isAdmin },
              { target: 'closed.error' },
            ],
          },
        },
        opened: {
          on: {
            CLOSE: 'closed',
          },
        },
      },
    };
    expect(xstateToSvg(description)).toMatchSnapshot();
  });
});
