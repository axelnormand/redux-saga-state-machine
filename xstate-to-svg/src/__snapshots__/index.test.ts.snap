// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`xstateToSvg ... 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"no\\"?>
<!DOCTYPE svg PUBLIC \\"-//W3C//DTD SVG 1.1//EN\\"
 \\"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\\">
<!-- Generated by graphviz version 2.40.1 (20161225.0304)
 -->
<!-- Title: state transitions Pages: 1 -->
<svg width=\\"71pt\\" height=\\"271pt\\"
 viewBox=\\"0.00 0.00 71.00 271.00\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\">
<g id=\\"graph0\\" class=\\"graph\\" transform=\\"scale(1 1) rotate(0) translate(4 267)\\">
<title>state transitions</title>
<polygon fill=\\"#ffffff\\" stroke=\\"transparent\\" points=\\"-4,4 -4,-267 67,-267 67,4 -4,4\\"/>
<!-- initial -->
<g id=\\"node1\\" class=\\"node\\">
<title>initial</title>
<ellipse fill=\\"#000000\\" stroke=\\"#000000\\" stroke-width=\\"2\\" cx=\\"31.5\\" cy=\\"-257.5\\" rx=\\"5.5\\" ry=\\"5.5\\"/>
</g>
<!-- backlog -->
<g id=\\"node2\\" class=\\"node\\">
<title>backlog</title>
<polygon fill=\\"#ffffff\\" stroke=\\"transparent\\" stroke-width=\\"2\\" points=\\"63,-204 0,-204 0,-168 63,-168 63,-204\\"/>
<text text-anchor=\\"start\\" x=\\"11.327\\" y=\\"-182.2\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"12.00\\" fill=\\"#000000\\">backlog</text>
<path fill=\\"none\\" stroke=\\"#000000\\" stroke-width=\\"2\\" d=\\"M12.3333,-169C12.3333,-169 50.6667,-169 50.6667,-169 56.3333,-169 62,-174.6667 62,-180.3333 62,-180.3333 62,-191.6667 62,-191.6667 62,-197.3333 56.3333,-203 50.6667,-203 50.6667,-203 12.3333,-203 12.3333,-203 6.6667,-203 1,-197.3333 1,-191.6667 1,-191.6667 1,-180.3333 1,-180.3333 1,-174.6667 6.6667,-169 12.3333,-169\\"/>
</g>
<!-- initial&#45;&gt;backlog -->
<g id=\\"edge1\\" class=\\"edge\\">
<title>initial&#45;&gt;backlog</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M31.5,-251.8895C31.5,-243.8095 31.5,-228.335 31.5,-214.4122\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"35.0001,-214.2159 31.5,-204.2159 28.0001,-214.2159 35.0001,-214.2159\\"/>
<text text-anchor=\\"middle\\" x=\\"32.8895\\" y=\\"-225\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\"> </text>
</g>
<!-- doing -->
<g id=\\"node3\\" class=\\"node\\">
<title>doing</title>
<polygon fill=\\"#ffffff\\" stroke=\\"transparent\\" stroke-width=\\"2\\" points=\\"59.5,-120 3.5,-120 3.5,-84 59.5,-84 59.5,-120\\"/>
<text text-anchor=\\"start\\" x=\\"16.827\\" y=\\"-98.2\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"12.00\\" fill=\\"#000000\\">doing</text>
<path fill=\\"none\\" stroke=\\"#000000\\" stroke-width=\\"2\\" d=\\"M15.8333,-85C15.8333,-85 47.1667,-85 47.1667,-85 52.8333,-85 58.5,-90.6667 58.5,-96.3333 58.5,-96.3333 58.5,-107.6667 58.5,-107.6667 58.5,-113.3333 52.8333,-119 47.1667,-119 47.1667,-119 15.8333,-119 15.8333,-119 10.1667,-119 4.5,-113.3333 4.5,-107.6667 4.5,-107.6667 4.5,-96.3333 4.5,-96.3333 4.5,-90.6667 10.1667,-85 15.8333,-85\\"/>
</g>
<!-- backlog&#45;&gt;doing -->
<g id=\\"edge2\\" class=\\"edge\\">
<title>backlog&#45;&gt;doing</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M31.5,-167.7733C31.5,-156.8758 31.5,-142.7946 31.5,-130.4587\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"35.0001,-130.1911 31.5,-120.1912 28.0001,-130.1912 35.0001,-130.1911\\"/>
<text text-anchor=\\"middle\\" x=\\"32.8895\\" y=\\"-141\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\"> </text>
</g>
<!-- test -->
<g id=\\"node4\\" class=\\"node\\">
<title>test</title>
<polygon fill=\\"#ffffff\\" stroke=\\"transparent\\" stroke-width=\\"2\\" points=\\"59.5,-36 3.5,-36 3.5,0 59.5,0 59.5,-36\\"/>
<text text-anchor=\\"start\\" x=\\"21.8298\\" y=\\"-14.2\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"12.00\\" fill=\\"#000000\\">test</text>
<path fill=\\"none\\" stroke=\\"#000000\\" stroke-width=\\"2\\" d=\\"M15.8333,-1C15.8333,-1 47.1667,-1 47.1667,-1 52.8333,-1 58.5,-6.6667 58.5,-12.3333 58.5,-12.3333 58.5,-23.6667 58.5,-23.6667 58.5,-29.3333 52.8333,-35 47.1667,-35 47.1667,-35 15.8333,-35 15.8333,-35 10.1667,-35 4.5,-29.3333 4.5,-23.6667 4.5,-23.6667 4.5,-12.3333 4.5,-12.3333 4.5,-6.6667 10.1667,-1 15.8333,-1\\"/>
</g>
<!-- doing&#45;&gt;test -->
<g id=\\"edge3\\" class=\\"edge\\">
<title>doing&#45;&gt;test</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M31.5,-83.7733C31.5,-72.8758 31.5,-58.7946 31.5,-46.4587\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"35.0001,-46.1911 31.5,-36.1912 28.0001,-46.1912 35.0001,-46.1911\\"/>
<text text-anchor=\\"middle\\" x=\\"32.8895\\" y=\\"-57\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\"> </text>
</g>
</g>
</svg>
"
`;
